qt/applications/workbench/workbench/plugins/editor.py

```diff
@@ -73,6 +76,22 @@ def app_closing(self):
         """
         return self.editors.close_all()
 
+    def dragEnterEvent(self, event):
+        data = event.mimeData()
+        if data.hasText() and data.hasUrls:
+            filepaths = [url.toLocalFile() for url in data.urls()]
+            for filepath in filepaths:
+                if filepath.endswith('.py') or filepath.endswith('.pyw'):
+                    event.acceptProposedAction()
+
+    def dropEvent(self, event):
+        data = event.mimeData()
+        for url in data.urls():
+            filepath = url.toLocalFile()
+            if filepath.endswith('.py') or filepath.endswith('.pyw'):
+                if osp.isfile(filepath):
```

Comment:
Files are always cause for concern regarding exceptions. Could we add a `try...except` around here and print out  a warning if an exception is raised?